src/Trigger/RabbitMQTriggerAttributeBindingProvider.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Triggers;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Extensions.RabbitMQ.Trigger
+{
+    internal class RabbitMQTriggerAttributeBindingProvider : ITriggerBindingProvider
+    {
+        private readonly INameResolver _nameResolver;
+        private readonly ILogger _logger;
+        private readonly IConfiguration _config;
+        private readonly IOptions<RabbitMQOptions> _options;
+        private readonly RabbitMQExtensionConfigProvider _provider;
+
+        public RabbitMQTriggerAttributeBindingProvider(
+            IConfiguration configuration,
+            INameResolver nameResolver,
+            IOptions<RabbitMQOptions> options,
+            RabbitMQExtensionConfigProvider provider,
+            ILoggerFactory loggerFactory)
+        {
+            _config = configuration;
+            _nameResolver = nameResolver;
+            _options = options;
+            _provider = provider;
+            _logger = loggerFactory?.CreateLogger(LogCategories.CreateTriggerCategory("RabbitMQ"));
+        }
+
+        public async Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext context)

```

Comment:
This never awaits because I never make any async calls in this function, but follows the programming model of other extensions. Should I still keep it as async?